### ISSUE TITLE
searchReferences only matches MD5 checksums when SearchReferencesRequest.md5checksums contains "TODO"

### ISSUE THREAD

**Issue by hjellinek** [ga4gh/server#540]
(The point of this issue is that the ref server needs to implement the MD5-related `Reference` search and retrieval operations.)

**Background**: I'm using data from the server unit tests as the server's data.  The reference data looks like this:
```
references/
    example_1/
        simple.fa.gz
        simple.fa.gz.fai
        simple.fa.gz.gzi
```

**Action**: I create a `SearchReferencesRequest` and set the value of `md5checksums` to contain the string `"TODO"` and pass it to `searchReferences`.

**Expected**:
The call to `searchReferences` will return a `SearchReferencesResponse` with its `references` containing zero results, since the `"TODO"` in the request is not a legal MD5 value.

**Actual**:
The call to `SearchReferencesResponse` returns a `SearchReferencesResponse` containing a single `Reference`.  The `Reference`'s `md5checksum` field contains the string __`"TODO"`__.

**Comment by jeromekelleher** [comment]
The issue here is that we haven't settled on an import format for references. We have the FASTA files with the data, but we need something else to go along with it to hold the metadata. This must also contain the md5 that we want. We can't compute the md5 on server startup, as references are large.

I think the best thing to do here is to make the md5 field an empty string, and raise a NotImplemented error when someone tries to search by md5 values. We should also create an issue for proper handling of references.

**Comment by jeromekelleher** [comment]
Fixed in #626.
